OAuth2 without service discovery: gateway compilation errors

##### **JHipster Version(s)**

```
gateway@0.0.0 /Users/alexandrastefanova/developer/projects/urban/gateway
└── generator-jhipster@6.0.0 

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.urban.gateway"
    },
    "jhipsterVersion": "6.0.0",
    "applicationType": "gateway",
    "baseName": "gateway",
    "packageName": "com.urban.gateway",
    "packageFolder": "com/urban/gateway",
    "serverPort": "8080",
    "authenticationType": "oauth2",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "clientFramework": "react",
    "clientTheme": "journal",
    "clientThemeVariant": "primary",
    "useSass": true,
    "clientPackageManager": "npm",
    "testFrameworks": [
      "gatling",
      "protractor"
    ],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": false
  }
}
</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Profile
relationship OneToOne {
  Profile{user(login)} to User
}

dto Profile with mapstruct
paginate Profile with infinite-scroll
service Profile with serviceClass
filter Profile

</pre>
</details>


##### **Environment and Tools**

java version "1.8.0_144"
Java(TM) SE Runtime Environment (build 1.8.0_144-b01)
Java HotSpot(TM) 64-Bit Server VM (build 25.144-b01, mixed mode)

git version 2.13.1

node: v10.15.1

npm: 6.9.0

yeoman: 2.0.5

yarn: 1.13.0

Docker version 18.09.2, build 6247962

docker-compose version 1.23.2, build 1110ad01

##### **Overview of the issue**

gateway process fails to compile class AuthorizationHeaderFilter with com.netflix.zuul not found error

##### **Motivation for or Use Case**

I want to deploy to a GCP kubernetes cluster with ingress/istio for load balancing.  So i choose to generate a jhipster gateway  without a jhipster registry 

##### **Reproduce the error**

generate a microservices gateway without a registry and with oauth2 authentication and compile it

##### **JHipster Version(s)**

6

##### **JHipster configuration**

{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.urban.gateway"
    },
    "jhipsterVersion": "6.0.0",
    "applicationType": "gateway",
    "baseName": "gateway",
    "packageName": "com.urban.gateway",
    "packageFolder": "com/urban/gateway",
    "serverPort": "8080",
    "authenticationType": "oauth2",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "clientFramework": "react",
    "clientTheme": "journal",
    "clientThemeVariant": "primary",
    "useSass": true,
    "clientPackageManager": "npm",
    "testFrameworks": [
      "gatling",
      "protractor"
    ],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": false
  }
}

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

profile.json

{
    "fluentMethods": true,
    "clientRootFolder": "",
    "relationships": [
        {
            "relationshipName": "user",
            "otherEntityName": "user",
            "relationshipType": "one-to-one",
            "otherEntityField": "login",
            "ownerSide": true,
            "useJPADerivedIdentifier": true,
            "otherEntityRelationshipName": "profile"
        }
    ],
    "fields": [],
    "changelogDate": "20190609151112",
    "dto": "mapstruct",
    "searchEngine": false,
    "service": "serviceClass",
    "entityTableName": "profile",
    "databaseType": "sql",
    "jpaMetamodelFiltering": true,
    "pagination": "infinite-scroll"
}

##### **Browsers and Operating System**

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

-   [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

As you choose "no service discovery", it's an option we didn't test in our CI.
If you find the fix, feel free to PR.

Here the full stacktrace for people who want to work on this:

```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/pgrimaud/tmp/13-issue-9877/src/main/java/com/urban/gateway/security/oauth2/AuthorizationHeaderFilter.java:[3,24] package com.netflix.zuul does not exist
[ERROR] /home/pgrimaud/tmp/13-issue-9877/src/main/java/com/urban/gateway/security/oauth2/AuthorizationHeaderFilter.java:[4,32] package com.netflix.zuul.context does not exist
[ERROR] /home/pgrimaud/tmp/13-issue-9877/src/main/java/com/urban/gateway/security/oauth2/AuthorizationHeaderFilter.java:[10,69] package org.springframework.cloud.netflix.zuul.filters.support does not exist
[ERROR] /home/pgrimaud/tmp/13-issue-9877/src/main/java/com/urban/gateway/security/oauth2/AuthorizationHeaderFilter.java:[10,1] static import only from classes and interfaces
[ERROR] /home/pgrimaud/tmp/13-issue-9877/src/main/java/com/urban/gateway/security/oauth2/AuthorizationHeaderFilter.java:[12,48] cannot find symbol
  symbol: class ZuulFilter
[INFO] 5 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  48.048 s
[INFO] Finished at: 2019-06-10T20:03:53+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project gateway: Compilation failure: Compilation failure: 
[ERROR] /home/pgrimaud/tmp/13-issue-9877/src/main/java/com/urban/gateway/security/oauth2/AuthorizationHeaderFilter.java:[3,24] package com.netflix.zuul does not exist
[ERROR] /home/pgrimaud/tmp/13-issue-9877/src/main/java/com/urban/gateway/security/oauth2/AuthorizationHeaderFilter.java:[4,32] package com.netflix.zuul.context does not exist
[ERROR] /home/pgrimaud/tmp/13-issue-9877/src/main/java/com/urban/gateway/security/oauth2/AuthorizationHeaderFilter.java:[10,69] package org.springframework.cloud.netflix.zuul.filters.support does not exist
[ERROR] /home/pgrimaud/tmp/13-issue-9877/src/main/java/com/urban/gateway/security/oauth2/AuthorizationHeaderFilter.java:[10,1] static import only from classes and interfaces
[ERROR] /home/pgrimaud/tmp/13-issue-9877/src/main/java/com/urban/gateway/security/oauth2/AuthorizationHeaderFilter.java:[12,48] cannot find symbol
[ERROR]   symbol: class ZuulFilter
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

--- a/generators/server/files.js
+++ b/generators/server/files.js
@@ -695,7 +695,7 @@ const serverFiles = {
         },
         {
             condition: generator =>
-                generator.authenticationType === 'oauth2' && generator.applicationType === 'gateway' && generator.serviceDiscoveryTypes,
+                generator.authenticationType === 'oauth2' && generator.applicationType === 'gateway' && generator.serviceDiscoveryType,
             path: SERVER_MAIN_SRC_DIR,
             templates: [
                 {

--- a/generators/server/files.js
+++ b/generators/server/files.js
@@ -694,7 +694,8 @@ const serverFiles = {
             ]
         },
         {
-            condition: generator => generator.authenticationType === 'oauth2' && generator.applicationType === 'gateway',
+            condition: generator =>
+                generator.authenticationType === 'oauth2' && generator.applicationType === 'gateway' && generator.serviceDiscoveryTypes,
             path: SERVER_MAIN_SRC_DIR,
             templates: [
                 {

--- a/generators/server/templates/src/main/java/package/config/SecurityConfiguration.java.ejs
+++ b/generators/server/templates/src/main/java/package/config/SecurityConfiguration.java.ejs
@@ -60,7 +60,7 @@ import <%=packageName%>.security.SecurityUtils;
 import org.springframework.security.oauth2.core.DelegatingOAuth2TokenValidator;
 import org.springframework.security.oauth2.core.OAuth2TokenValidator;
 import org.springframework.security.oauth2.jwt.*;
-        <%_ if (applicationType === 'gateway') { _%>
+        <%_ if (applicationType === 'gateway' && serviceDiscoveryType) { _%>
 import <%=packageName%>.security.oauth2.AuthorizationHeaderFilter;
 import <%=packageName%>.security.oauth2.AuthorizationHeaderUtil;
         <%_ } _%>
@@ -330,7 +330,7 @@ public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
 
         return jwtDecoder;
     }
-        <%_ if (applicationType === 'gateway') { _%>
+        <%_ if (applicationType === 'gateway' && serviceDiscoveryType) { _%>
 
     @Bean
     public AuthorizationHeaderFilter authHeaderFilter(AuthorizationHeaderUtil headerUtil) {

